app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/OutcomeValueFunction.scala

```diff
@@ -0,0 +1,255 @@
+package org.bitcoins.commons.jsonmodels.dlc
+
+import org.bitcoins.core.currency.Satoshis
+import org.bitcoins.core.util.NumberUtil
+
+import scala.math.BigDecimal.RoundingMode
+
+/** A DLC payout curve defined by piecewise interpolating points */
+case class OutcomeValueFunction(points: Vector[OutcomeValuePoint]) {
+
+  /** These points (and their indices in this.points) represent the endpoints
+    * between which interpolation happens.
+    * In other words these endpoints define the pieces of the piecewise function.
+    */
+  lazy val endpoints: Vector[(OutcomeValuePoint, Int)] =
+    points.zipWithIndex.filter(_._1.isEndpoint)
+
+  /** This Vector contains the function pieces between the endpoints */
+  lazy val functionComponents: Vector[OutcomeValueFunctionComponent] = {
+    endpoints.init.zip(endpoints.tail).map { // All pairs of adjacent endpoints
+      case ((_, index), (_, nextIndex)) =>
+        OutcomeValueFunctionComponent(points.slice(index, nextIndex + 1))
+    }
+  }
+
+  private lazy val outcomes = endpoints.map(_._1.outcome)
+
+  def componentFor(
+      outcome: BigDecimal): (OutcomeValueFunctionComponent, Int) = {
+    val endpointIndex = NumberUtil.search(outcomes, outcome)
+    val (endpoint, _) = endpoints(endpointIndex)
+
+    if (
+      endpoint.outcome == outcome && endpointIndex != functionComponents.length
+    ) {
+      (functionComponents(endpointIndex), endpointIndex)
+    } else {
+      (functionComponents(endpointIndex - 1), endpointIndex - 1)
+    }
+  }
+
+  def apply(outcome: BigDecimal): Satoshis = {
+    val (func, _) = componentFor(outcome)
+    func(outcome)
+  }
+
+  def apply(outcome: BigDecimal, rounding: RoundingIntervals): Satoshis = {
+    val (func, _) = componentFor(outcome)
+    func(outcome, rounding)
+  }
+}
+
+/** A point on a DLC payout curve to be used for interpolation
+  *
+  * @param outcome An element of the domain of possible events signed by the oracle
+  * @param value The payout to the local party corresponding to outcome
+  * @param isEndpoint True if this point defines a boundary between pieces in the curve
+  */
+case class OutcomeValuePoint(
+    outcome: BigDecimal,
+    value: Satoshis,
+    isEndpoint: Boolean)
```

Comment:
I'm not sure I follow, why would changing `isEndpoint` to a type cause a compiler error?

Reply:
oooh, you're talking about making `OutcomeValuePoint` an ADT I think?

Reply:
I'll add it to the list for subsequent PR

Reply:
That's totally cool with me, but can we get issues filed? Otherwise I know I will forget. 

Some of this stuff is just boiler plate software stuff, so Ben or I can start peeling these off the backlog if we have documentation of it :+1: